test/app-luatest/gh_7808_rel_paths_in_diag_and_debug_info_test.lua

```diff
@@ -0,0 +1,25 @@
+local bit = require('bit')
+local misc = require('test.luatest_helpers.misc')
+local netbox = require('net.box')
+local t = require('luatest')
+
+local g = t.group()
+
+g.test_rel_paths_in_diag = function(_)
+    local _, err = pcall(function() netbox.self:call('') end)
+    t.assert_not(err.trace[1].file:startswith('/'))
+end
+
+g.test_rel_paths_in_debug_info = function(_)
+    misc.skip_if_macOS()
+
+    local readelf = ("readelf --debug-dump=info %s"):format(arg[-1])
+    local grep_comp_dir = "grep DW_AT_comp_dir"
+    local awk = "awk '{print $9}'"
+    local grep_abs_path = "grep '^/'"
+    local rc = os.execute(table.concat({readelf, grep_comp_dir, awk, grep_abs_path}, '|'))
```

Comment:
Nit: I'd use `popen` here, because it'd allow for better error reporting (we could print which paths are not relative). Up to @Totktonada.

Reply:
`popen` does not have convenient helpers for reading.